File: typescript/src/patterns/cqrs/WriteModel.ts
import type { DomainEvent } from './Events.js';
import { EventBus } from './EventBus.js';

export class WriteModel {
  constructor(private bus: EventBus) {}
  createUser(id: string, name: string, email: string) {
    const evt: DomainEvent = { type: 'UserCreated', id, name, email };
    this.bus.publish(evt);
  }
  updateUserName(id: string, name: string) {
    const evt: DomainEvent = { type: 'UserNameUpdated', id, name };
    this.bus.publish(evt);
  }
}

File: python/src/patterns/cqrs/write_model.py
from .event_bus import EventBus
from .events import UserCreated, UserNameUpdated

class WriteModel:
    def __init__(self, bus: EventBus) -> None:
        self.bus = bus
    def create_user(self, id: str, name: str, email: str) -> None:
        self.bus.publish(UserCreated(id, name, email))
    def update_user_name(self, id: str, name: str) -> None:
        self.bus.publish(UserNameUpdated(id, name))
